plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -4729,6 +4729,17 @@ private VirtualMachineGuestOsIdentifier translateGuestOsIdentifier(String cpuArc
         return VirtualMachineGuestOsIdentifier.OTHER_GUEST;
     }
 
+    private String translateGuestOsIdentifierEx(String cpuArchitecture, String guestOs, String cloudGuestOs) {
+        if (cloudGuestOs != null) {
```

Comment:
agreed with @rafaelweingartner -- short circuiting out of the method greatly reduces complexity by reducing nesting.

Also, it seems like the check should be ``Strings.isBlankorNull`` to check for empty and, as well as, null values.